packages/plugin-core/src/features/codeActionProvider.ts

```diff
@@ -263,6 +277,11 @@ export const refactorProvider: CodeActionProvider = {
         // return if none
         return;
       } else {
+        //regex for url
+        const regex = "^(http://www.|https://www.|http://|https://)?[a-z0-9]+([-.]{1}[a-z0-9]+)*.[a-z]{2,5}(:[0-9]{1,5})?(/.*| [^ ]*)?$";
```

Comment:
That's one heavy regex :D 

Can we have a link to https://regex101.com/ with few examples of it matching, or the source of this regex if its copied from somewhere (or both)

Reply:
Heads up that this doesn't match capitalized titles like `http://Google.com`, or local urls like `http://localhost:3000`, and internationalized domain names like `http://xn--bcher-kva.de/` (see https://unicode.org/faq/idn.html).

Reply:
used [is-Url](https://www.npmjs.com/package/is-url)